extractor/extractor/filing.py

```diff
@@ -0,0 +1,135 @@
+from __future__ import annotations
+import logging
+from lxml import etree
+import sys
+from typing import Dict, Optional, TextIO
+
+
+_logger = logging.getLogger(__name__)
+
+
+class Filing:
+    """
+    Encapsulate a Filing as a semantic object with appropriately named
+    fields.
+
+    The implementation actually returned from Result objects will have its
+    fields generated at build time to avoid the need to update the code every
+    time someone wants to access a new field.
+
+    Fields that aren't included can still be accessed using the XPath strings
+    listed on http://www.irsx.info/ and the `get_*` methods. The paths should
+    be used verbatim, in their absolute form, without namespaces.
+
+    >>> with open('fixtures/filing.xml') as xml:
+    ...    filing = Filing(xml)
+    >>> filing.formation_year
+    '2004'
+    >>> filing.principal_officer_name
+    'SCOTT LEWIS'
+    """
+
+    _namespaces: Dict[str, str]
+
+    _tree: etree.ElementTree
+
+    def __init__(
+        self, xml_file: TextIO,
+    ):
+        self._namespaces = {
+            "efile": "http://www.irs.gov/efile",
+        }
+
+        content = xml_file.read()
+        while not content.startswith("<"):
+            content = content[1:]
+        content = content.encode("utf-8")
+        _logger.info(f"XML: '{content[:40]}'")
+
+        self._tree = etree.fromstring(content)
+    
+    @property
+    def activity_or_mission_description(self) -> Optional[str]:
+        return self.get_str("/IRS990/ActivityOrMissionDesc")
+    
+    @property
+    def formation_year(self) -> Optional[str]:
+        return self.get_str("/IRS990/FormationYr")
+    
+    @property
+    def gross_receipts(self) -> Optional[int]:
+        return self.get_int("/IRS990/GrossReceiptsAmt")
+    
+    @property
+    def employee_count(self) -> Optional[int]:
+        return self.get_int("/IRS990/TotalEmployeeCnt")
+
+    @property
+    def principal_officer_name(self) -> Optional[str]:
+        return self.get_str("/IRS990/PrincipalOfficerNm")
+    
+    @property
+    def us_address(self) -> Optional[str]:
```

Comment:
possible TODO - would it be up in here that we'd add the conversion of the address to long/lat to a field if it's not already populated?

Reply:
Yeah, good call, the lat/lon stuff will need to plug in somewhere around here.